### PR TITLE
chore(types): fix more typescript types

### PR DESCRIPTION
**Pull request by viceice**
* makes `RenovateConfig` stricter
* fixes and unifies a lot of types

rel: #1261

### CODE REVIEW

File: lib/workers/pr/body/common.ts [hunk 1]
```diff
@@ -14,16 +14,18 @@ export type PrBodyUpgrade = LookupUpdate &
...
+//   prBodyTemplate: string;
+//   prBodyColumns: string[];
+//   prBodyDefinitions: Record<string, string>;
+//   prTitle?: string;
+//   updateType: UpdateType;
+//   isGroup?: boolean;
+//   isPin?: boolean;
+//   hasReleaseNotes?: boolean;
+//   global?: GlobalConfig;
+// } & Partial<RenovateSharedConfig>;
```
Comment (rarkins) [inline lib/workers/pr/body/common.ts]: Delete?

**Review by rarkins — DISMISSED** [review]
Looks good, maybe one deletion of commented out code

**Comment by JamieMagee** [comment]
Nice! I've got some changes around cleaning up types in Azure DevOps, but I'll let you merge this first to avoid conflicts.

**Review by JamieMagee — DISMISSED** [review]

File: lib/workers/pr/body/common.ts (hunk 1 above)
Comment (viceice) [reply lib/workers/pr/body/common.ts]: ups, sure. maybe i can remove some more obsolete interfaces and reexported type declarations

**Comment by rarkins** [comment]
@viceice let me know when good to merge

**Comment by viceice** [comment]
@rarkins done

**Pull request merged**